src/services/event-service.js

```diff
@@ -1,5 +1,21 @@
+import http from './http-service';
+
+const resourceName = 'events';
+
 export default {
-  getEvent(id) {
-    return { id };
+  get() {
+    return http.get(`/${resourceName}`);
+  },
+  getById(id) {
+    return http.get(`/${resourceName}/${id}`);
+  },
+  create(item) {
+    return http.post(`/${resourceName}`, item);
+  },
+  update(item) {
+    return http.put(`/${resourceName}/${item.id}`, item);
+  },
```

Comment:
data-provider was incomplete. In my opinion, this way to create a DataService is centralizing this logic and is using axios interceptors to manage requests and responses in a better way.

Reply:
Axios interceptors will not be used because if the data library is changed (E.G. customer decides to use graphQL) the logic inside the interceptors would be migrated to, is better manage that part in the state layer. Please change it by the data provider.